cumulusci/utils/parallel/task_worker_queues/parallel_worker.py

```diff
@@ -135,20 +135,22 @@ def run(self):
                 self.subtask()
                 logger.info(str(self.subtask.return_values))
                 logger.info("SubTask Success!")
-                self.results_reporter.put(
-                    {
-                        "status": "success",
-                        "results": self.subtask.return_values,
-                        "directory": str(self.working_dir),
-                    }
-                )
+                if self.results_reporter:
+                    self.results_reporter.put(
+                        {
+                            "status": "success",
+                            "results": self.subtask.return_values,
+                            "directory": str(self.working_dir),
+                        }
+                    )
             except BaseException as e:
                 logger.info(f"Failure detected: {e}")
                 self.save_exception(e)
                 self.failures_dir.mkdir(exist_ok=True)
                 logfile.close()
                 shutil.move(str(self.working_dir), str(self.failures_dir))
-                self.results_reporter.put({"status": "error", "error": str(e)})
+                if self.results_reporter:
+                    self.results_reporter.put({"status": "error", "error": str(e)})
```

Comment:
It's fine as is, but you could avoid all these conditionals by implementing a NullResultsReporter that doesn't actually do anything with the messages passed to it

Reply:
I did debate that internally but I thought it might be equally confusing if someone was reading the code and it looked like something was being put into the queue but it wasn't "appearing" anywhere. Since there are only two (?) places with the conditional, I thought it was clearer this way.